32-bit build failure with 2.5.3 -> 2.5.4

### Checklist

- [x] I'm reporting a problem with Chatterino
- [x] I've verified that I'm running **the most** recent nightly build or stable release
- [x] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [x] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

Hi, Alpine Linux maintainer here. I was upgrading our chatterino package from 2.5.3 to 2.5.4 today and noticed that on 32-bit architectures (for us: armhf, armv7, x86) the following build failure:

```
ninja: job failed: /usr/bin/c++ -DAB_CUSTOM_SETTINGS -DCHATTERINO -DCHATTERINO_DISABLE_UPDATER -DCHATTERINO_HAVE_PLUGINS -DCHATTERINO_WITH_LIBNOTIFY -DCMAKE_BUILD -DIRC_NAMESPACE=Communi -DIRC_STATIC -DQT_CONCURRENT_LIB -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x060403 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_NO_KEYWORDS -DQT_SVG_LIB -DQT_WARN_DEPRECATED_UP_TO=0x060403 -DQT_WIDGETS_LIB -DSOL_ALL_SAFETIES_ON=1 -DSOL_NO_NIL=0 -DSOL_USING_CXX_LUA=1 -DUNICODE -I/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/build/src/chatterino-lib_autogen/include -I/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src -I/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/build/autogen -I/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/semver/include -I/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/expected-lite/include -I/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/miniaudio -I/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/libcommuni/include/IrcCore -I/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/libcommuni/include/IrcModel -I/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/libcommuni/include/IrcUtil -I/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/settings/include -I/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/twitch-eventsub-ws/src/../include -I/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/lua/src -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtConcurrent -isystem /usr/include/qt6/QtCore5Compat -isystem /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/serialize/include -isystem /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/signals/include -isystem /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/websocketpp -isystem /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/lrucache -isystem /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/magic_enum/include -isystem /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/certify/include -isystem /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/sol2/include -isystem /usr/include/qt6/QtDBus -isystem /usr/include/gdk-pixbuf-2.0 -isystem /usr/include/libpng16 -isystem /usr/include/webp -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/glib-2.0 -isystem /usr/lib/glib-2.0/include -Os -fstack-clash-protection -Wformat -Werror=format-security -D_GLIBCXX_ASSERTIONS=1 -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS=1 -D_LIBCPP_ENABLE_HARDENED_MODE=1 -std=gnu++20 -Wall -Wno-unused-function -Wno-switch -Wno-deprecated-declarations -Wno-sign-compare -Wno-strict-aliasing -Werror=return-type -Werror=reorder -Wno-class-memaccess -Wno-unused-variable -pthread -MD -MT src/CMakeFiles/chatterino-lib.dir/widgets/PluginRepl.cpp.o -MF src/CMakeFiles/chatterino-lib.dir/widgets/PluginRepl.cpp.o.d -o src/CMakeFiles/chatterino-lib.dir/widgets/PluginRepl.cpp.o -c /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/widgets/PluginRepl.cpp
/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/widgets/PluginRepl.cpp: In member function 'std::optional<QString> {anonymous}::HistoricTextEdit::nextHistoryItem(qsizetype)':
/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/widgets/PluginRepl.cpp:151:19: error: no matching function for call to 'clamp(qsizetype, long long int, qsizetype)'
  151 |         std::clamp(this->historyIdx + diff, 0LL, this->history.size());
      |         ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/widgets/PluginRepl.cpp:151:19: note: there are 2 candidates
In file included from /usr/include/c++/15.2.0/algorithm:63,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/signals/include/pajlada/signals/signal.hpp:5,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/widgets/BaseWidget.hpp:3,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/widgets/buttons/Button.hpp:3,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/widgets/buttons/SvgButton.hpp:3,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/widgets/PluginRepl.hpp:4,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/widgets/PluginRepl.cpp:1:
/usr/include/c++/15.2.0/bits/stl_algo.h:3636:5: note: candidate 1: 'template<class _Tp> constexpr const _Tp& std::clamp(const _Tp&, const _Tp&, const _Tp&)'
 3636 |     clamp(const _Tp& __val, const _Tp& __lo, const _Tp& __hi)
      |     ^~~~~
/usr/include/c++/15.2.0/bits/stl_algo.h:3636:5: note: template argument deduction/substitution failed:
/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/widgets/PluginRepl.cpp:151:19: note:   deduced conflicting types for parameter 'const _Tp' ('int' and 'long long int')
  151 |         std::clamp(this->historyIdx + diff, 0LL, this->history.size());
      |         ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/15.2.0/bits/stl_algo.h:3656:5: note: candidate 2: 'template<class _Tp, class _Compare> constexpr const _Tp& std::clamp(const _Tp&, const _Tp&, const _Tp&, _Compare)'
 3656 |     clamp(const _Tp& __val, const _Tp& __lo, const _Tp& __hi, _Compare __comp)
      |     ^~~~~
/usr/include/c++/15.2.0/bits/stl_algo.h:3656:5: note: candidate expects 4 arguments, 3 provided
In file included from /usr/include/qt6/QtCore/qjsonobject.h:8,
                 from /usr/include/qt6/QtCore/QJsonObject:1,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/util/ImageUploader.cpp:6:
In member function 'QString QJsonValueConstRef::objectKey() const',
    inlined from 'QString QJsonObject::const_iterator::key() const' at /usr/include/qt6/QtCore/qjsonobject.h:218:60,
    inlined from 'QStringList chatterino::imageuploader::detail::{anonymous}::parseHeaders(const QJsonObject&)' at /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/util/ImageUploader.cpp:77:45,
    inlined from 'bool chatterino::imageuploader::detail::importSettings(const QJsonObject&, chatterino::Settings&)' at /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/util/ImageUploader.cpp:145:77:
/usr/include/qt6/QtCore/qjsonvalue.h:203:55: note: parameter passing for argument of type 'const QJsonValueConstRef' changed in GCC 9.1
  203 |     QString objectKey() const { return objectKey(*this); }
      |                                                       ^
In member function 'QString QJsonValueConstRef::toString(const QString&) const',
    inlined from 'QStringList chatterino::imageuploader::detail::{anonymous}::parseHeaders(const QJsonObject&)' at /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/util/ImageUploader.cpp:77:68,
    inlined from 'bool chatterino::imageuploader::detail::importSettings(const QJsonObject&, chatterino::Settings&)' at /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/util/ImageUploader.cpp:145:77:
/usr/include/qt6/QtCore/qjsonvalue.h:168:48: note: parameter passing for argument of type 'QJsonValueConstRef' changed in GCC 9.1
  168 |     { return concreteString(*this, defaultValue); }
      |                                                ^
/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/singletons/StreamerMode.cpp:156:56: warning: missing terminating ' character
  156 | #    warning Unsupported OS: Broadcasting software can\'t be detected
      |                                                        ^
In file included from /usr/include/qt6/QtCore/qjsonarray.h:8,
                 from /usr/include/qt6/QtCore/QJsonArray:1,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/singletons/NativeMessaging.cpp:17:
In member function 'QString QJsonValueConstRef::toString(const QString&) const',
    inlined from 'void chatterino::NativeMessagingServer::syncChannels(const QJsonArray&)' at /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/singletons/NativeMessaging.cpp:411:36:
/usr/include/qt6/QtCore/qjsonvalue.h:168:48: note: parameter passing for argument of type 'const QJsonValueConstRef' changed in GCC 9.1
  168 |     { return concreteString(*this, defaultValue); }
      |                                                ^
/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/singletons/NativeMessaging.cpp: In function 'void chatterino::registerNmHost(const Paths&)':
/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/singletons/NativeMessaging.cpp:155:58: note: parameter passing for argument of type 'std::initializer_list<QJsonValue>' changed in GCC 7.1
  155 |             u"chrome-extension://%1/"_s.arg(EXTENSION_ID)};
      |                                                          ^
/builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/singletons/NativeMessaging.cpp:175:75: note: parameter passing for argument of type 'std::initializer_list<QJsonValue>' changed in GCC 7.1
  175 |         QJsonArray allowedExtensions = {"chatterino_native@chatterino.com"};
      |                                                                           ^
In file included from /usr/include/c++/15.2.0/vector:67,
                 from /usr/include/c++/15.2.0/functional:66,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/lib/magic_enum/include/magic_enum/magic_enum.hpp:42,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/util/QMagicEnum.hpp:3,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/common/ChatterinoSetting.hpp:3,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/util/SignalListener.hpp:3,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/singletons/WindowManager.hpp:4,
                 from /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/singletons/WindowManager.cpp:1:
/usr/include/c++/15.2.0/bits/stl_uninitialized.h: In function 'constexpr _ForwardIterator std::__do_uninit_copy(_InputIterator, _Sentinel, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, vector<variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >; _Sentinel = __gnu_cxx::__normal_iterator<const variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, vector<variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >; _ForwardIterator = variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*]':
/usr/include/c++/15.2.0/bits/stl_uninitialized.h:140:5: note: parameter passing for argument of type '__gnu_cxx::__normal_iterator<const std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, std::vector<std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >' changed in GCC 7.1
  140 |     __do_uninit_copy(_InputIterator __first, _Sentinel __last,
      |     ^~~~~~~~~~~~~~~~
/usr/include/c++/15.2.0/bits/stl_uninitialized.h:140:5: note: parameter passing for argument of type '__gnu_cxx::__normal_iterator<const std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, std::vector<std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >' changed in GCC 7.1
In function '_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, vector<variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >; _ForwardIterator = variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*]',
    inlined from 'constexpr _ForwardIterator std::__uninitialized_copy_a(_InputIterator, _Sentinel, _ForwardIterator, allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<const variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, vector<variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >; _Sentinel = __gnu_cxx::__normal_iterator<const variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, vector<variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >; _ForwardIterator = variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*; _Tp = variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>]' at /usr/include/c++/15.2.0/bits/stl_uninitialized.h:635:32,
    inlined from 'constexpr std::vector<_Tp, _Alloc>::vector(const std::vector<_Tp, _Alloc>&) [with _Tp = std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>; _Alloc = std::allocator<std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> >]' at /usr/include/c++/15.2.0/bits/stl_vector.h:636:31:
/usr/include/c++/15.2.0/bits/stl_uninitialized.h:266:37: note: parameter passing for argument of type '__gnu_cxx::__normal_iterator<const std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, std::vector<std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >' changed in GCC 7.1
  266 |         return std::__do_uninit_copy(__first, __last, __result);
      |                ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
In function '_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, vector<variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >; _ForwardIterator = variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*]',
    inlined from 'constexpr _ForwardIterator std::__uninitialized_copy_a(_InputIterator, _Sentinel, _ForwardIterator, allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<const variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, vector<variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >; _Sentinel = __gnu_cxx::__normal_iterator<const variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, vector<variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >; _ForwardIterator = variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*; _Tp = variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>]' at /usr/include/c++/15.2.0/bits/stl_uninitialized.h:635:32,
    inlined from 'constexpr std::vector<_Tp, _Alloc>::pointer std::vector<_Tp, _Alloc>::_M_allocate_and_copy(size_type, _ForwardIterator, _ForwardIterator) [with _ForwardIterator = __gnu_cxx::__normal_iterator<const std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, std::vector<std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >; _Tp = std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>; _Alloc = std::allocator<std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> >]' at /usr/include/c++/15.2.0/bits/stl_vector.h:1912:6,
    inlined from 'constexpr std::vector<_Tp, _Alloc>& std::vector<_Tp, _Alloc>::operator=(const std::vector<_Tp, _Alloc>&) [with _Tp = std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>; _Alloc = std::allocator<std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> >]' at /usr/include/c++/15.2.0/bits/vector.tcc:238:44,
    inlined from 'constexpr chatterino::ContainerNodeDescriptor& chatterino::ContainerNodeDescriptor::operator=(const chatterino::ContainerNodeDescriptor&)' at /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/common/WindowDescriptors.hpp:63:8,
    inlined from 'std::__detail::__variant::_Copy_assign_base<false, chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>::operator=(const std::__detail::__variant::_Copy_assign_base<false, chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>&)::<lambda(auto:20&&, auto:21)> mutable [with auto:20 = const chatterino::ContainerNodeDescriptor&; auto:21 = std::integral_constant<unsigned int, 0>]' at /usr/include/c++/15.2.0/variant:668:37,
    inlined from 'constexpr _Res std::__invoke_impl(__invoke_other, _Fn&&, _Args&& ...) [with _Res = void; _Fn = __detail::__variant::_Copy_assign_base<false, chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>::operator=(const std::__detail::__variant::_Copy_assign_base<false, chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>&)::<lambda(auto:20&&, auto:21)>; _Args = {const chatterino::ContainerNodeDescriptor&, integral_constant<unsigned int, 0>}]' at /usr/include/c++/15.2.0/bits/invoke.h:63:36,
    inlined from 'constexpr typename std::__invoke_result<_Functor, _ArgTypes>::type std::__invoke(_Callable&&, _Args&& ...) [with _Callable = __detail::__variant::_Copy_assign_base<false, chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>::operator=(const std::__detail::__variant::_Copy_assign_base<false, chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>&)::<lambda(auto:20&&, auto:21)>; _Args = {const chatterino::ContainerNodeDescriptor&, integral_constant<unsigned int, 0>}]' at /usr/include/c++/15.2.0/bits/invoke.h:98:40,
    inlined from 'static constexpr decltype(auto) std::__detail::__variant::__gen_vtable_impl<std::__detail::__variant::_Multi_array<_Result_type (*)(_Visitor, _Variants ...)>, std::integer_sequence<unsigned int, __indices ...> >::__visit_invoke(_Visitor&&, _Variants ...) [with _Result_type = std::__detail::__variant::__variant_idx_cookie; _Visitor = std::__detail::__variant::_Copy_assign_base<false, chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>::operator=(const std::__detail::__variant::_Copy_assign_base<false, chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>&)::<lambda(auto:20&&, auto:21)>&&; _Variants = {const std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>&}; unsigned int ...__indices = {0}]' at /usr/include/c++/15.2.0/variant:[1044](https://gitlab.alpinelinux.org/alpine/aports/-/jobs/2023495#L1044):17,
    inlined from 'constexpr decltype(auto) std::__do_visit(_Visitor&&, _Variants&& ...) [with _Result_type = __detail::__variant::__variant_idx_cookie; _Visitor = __detail::__variant::_Copy_assign_base<false, chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>::operator=(const std::__detail::__variant::_Copy_assign_base<false, chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>&)::<lambda(auto:20&&, auto:21)>; _Variants = {const variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>&}]' at /usr/include/c++/15.2.0/variant:1891:5,
    inlined from 'constexpr void std::__detail::__variant::__raw_idx_visit(_Visitor&&, _Variants&& ...) [with _Visitor = _Copy_assign_base<false, chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>::operator=(const std::__detail::__variant::_Copy_assign_base<false, chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>&)::<lambda(auto:20&&, auto:21)>; _Variants = {const std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>&}]' at /usr/include/c++/15.2.0/variant:187:44,
    inlined from 'constexpr std::__detail::__variant::_Copy_assign_base<<anonymous>, _Types>& std::__detail::__variant::_Copy_assign_base<<anonymous>, _Types>::operator=(const std::__detail::__variant::_Copy_assign_base<<anonymous>, _Types>&) [with bool <anonymous> = false; _Types = {chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor}]' at /usr/include/c++/15.2.0/variant:661:28:
/usr/include/c++/15.2.0/bits/stl_uninitialized.h:266:37: note: parameter passing for argument of type '__gnu_cxx::__normal_iterator<const std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor>*, std::vector<std::variant<chatterino::ContainerNodeDescriptor, chatterino::SplitNodeDescriptor> > >' changed in GCC 7.1
  266 |         return std::__do_uninit_copy(__first, __last, __result);
      |                ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/15.2.0/bits/stl_uninitialized.h: In function 'constexpr _ForwardIterator std::__do_uninit_copy(_InputIterator, _Sentinel, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, vector<chatterino::TabDescriptor> >; _Sentinel = __gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, vector<chatterino::TabDescriptor> >; _ForwardIterator = chatterino::TabDescriptor*]':
/usr/include/c++/15.2.0/bits/stl_uninitialized.h:140:5: note: parameter passing for argument of type '__gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, std::vector<chatterino::TabDescriptor> >' changed in GCC 7.1
  140 |     __do_uninit_copy(_InputIterator __first, _Sentinel __last,
      |     ^~~~~~~~~~~~~~~~
/usr/include/c++/15.2.0/bits/stl_uninitialized.h:140:5: note: parameter passing for argument of type '__gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, std::vector<chatterino::TabDescriptor> >' changed in GCC 7.1
In function '_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, vector<chatterino::TabDescriptor> >; _ForwardIterator = chatterino::TabDescriptor*]',
    inlined from 'constexpr _ForwardIterator std::__uninitialized_copy_a(_InputIterator, _Sentinel, _ForwardIterator, allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, vector<chatterino::TabDescriptor> >; _Sentinel = __gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, vector<chatterino::TabDescriptor> >; _ForwardIterator = chatterino::TabDescriptor*; _Tp = chatterino::TabDescriptor]' at /usr/include/c++/15.2.0/bits/stl_uninitialized.h:635:32,
    inlined from 'constexpr std::vector<_Tp, _Alloc>::vector(const std::vector<_Tp, _Alloc>&) [with _Tp = chatterino::TabDescriptor; _Alloc = std::allocator<chatterino::TabDescriptor>]' at /usr/include/c++/15.2.0/bits/stl_vector.h:636:31,
    inlined from 'constexpr chatterino::WindowDescriptor::WindowDescriptor(const chatterino::WindowDescriptor&)' at /builds/alpine/aports/community/chatterino2/src/chatterino2-2.5.4/src/common/WindowDescriptors.hpp:82:8:
/usr/include/c++/15.2.0/bits/stl_uninitialized.h:266:37: note: parameter passing for argument of type '__gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, std::vector<chatterino::TabDescriptor> >' changed in GCC 7.1
  266 |         return std::__do_uninit_copy(__first, __last, __result);
      |                ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
In function '_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, vector<chatterino::TabDescriptor> >; _ForwardIterator = chatterino::TabDescriptor*]',
    inlined from 'constexpr _ForwardIterator std::__uninitialized_copy_a(_InputIterator, _Sentinel, _ForwardIterator, allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, vector<chatterino::TabDescriptor> >; _Sentinel = __gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, vector<chatterino::TabDescriptor> >; _ForwardIterator = chatterino::TabDescriptor*; _Tp = chatterino::TabDescriptor]' at /usr/include/c++/15.2.0/bits/stl_uninitialized.h:635:32,
    inlined from 'constexpr std::vector<_Tp, _Alloc>::pointer std::vector<_Tp, _Alloc>::_M_allocate_and_copy(size_type, _ForwardIterator, _ForwardIterator) [with _ForwardIterator = __gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, std::vector<chatterino::TabDescriptor> >; _Tp = chatterino::TabDescriptor; _Alloc = std::allocator<chatterino::TabDescriptor>]' at /usr/include/c++/15.2.0/bits/stl_vector.h:1912:6,
    inlined from 'constexpr std::vector<_Tp, _Alloc>& std::vector<_Tp, _Alloc>::operator=(const std::vector<_Tp, _Alloc>&) [with _Tp = chatterino::TabDescriptor; _Alloc = std::allocator<chatterino::TabDescriptor>]' at /usr/include/c++/15.2.0/bits/vector.tcc:238:44:
/usr/include/c++/15.2.0/bits/stl_uninitialized.h:266:37: note: parameter passing for argument of type '__gnu_cxx::__normal_iterator<const chatterino::TabDescriptor*, std::vector<chatterino::TabDescriptor> >' changed in GCC 7.1
  266 |         return std::__do_uninit_copy(__first, __last, __result);
      |                ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: subcommand failed
```

If you wish to not support 32-bit architectures and rather let me disable the builds for them, thats fine.

But in case you're interested in fixing this, I could also try to get more information like bisecting the commit or so. I also believe this should be easily reproducible on any 32-bit system (if not just running a docker container).

### Screenshots

_No response_

### OS and Chatterino Version

Chatterino 2.5.4, up-to-date Alpine Linux edge (Qt 6.9.2)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,7 @@
 ## Unversioned
 
 - Dev: Make code sanitizers opt in with the `CHATTERINO_SANITIZER_SUPPORT` CMake option. After that's enabled, use the `SANITIZE_*` flag to enable individual sanitizers. (#6493)
+- Dev: Fix 32-bit compile in PluginRepl. (#6483)
 
 ## 2.5.4
 

--- a/src/widgets/PluginRepl.cpp
+++ b/src/widgets/PluginRepl.cpp
@@ -148,7 +148,7 @@ std::optional<QString> HistoricTextEdit::nextHistoryItem(qsizetype diff)
     bool wasUnfinishedInput = this->historyIdx >= this->history.size();
 
     auto nextIdx =
-        std::clamp(this->historyIdx + diff, 0LL, this->history.size());
+        std::clamp<qsizetype>(this->historyIdx + diff, 0, this->history.size());
     if (nextIdx == this->historyIdx)
     {
         return {};  // nothing changed

